### PR TITLE
Feature/delete dish

### PR DESCRIPTION
**Pull request by valdrinkoshi**
Currently BE side allows unconditional delete of dishes, but this PR handles also the case when we have an error, notifying the user

### CODE REVIEW

File: app/scripts/dishes.controller.js [hunk 1]
```diff
@@ -1,54 +1,73 @@
...
-    var that = this;
-
-    that.dishes = [];
-    that.modal = null;
-
-    that.hideModal = function() {
-      that.modal.hide();
-    };
-    that.addDish = function(dish, form) {
+(function () {
```
Comment (dbaq) [inline app/scripts/dishes.controller.js]: ```'use strict';``` should be inside the IIFE.

File: app/scripts/hc-messaging.service.js [hunk 2]
```diff
@@ -1,21 +1,34 @@
...
-          '<pre ng-show="showmore" style="font-size: x-small"><code>' +
-          JSON.stringify(error, null, '  ') + '</code></pre>'
-        });
-      };
-      return {
-        showError: showError
-      };
-    }]
-);
+(function () {
```
Comment (dbaq) [inline app/scripts/hc-messaging.service.js]: same here for use strict

File: app/scripts/dishes.controller.js [hunk 3]
```diff
@@ -1,54 +1,73 @@
...
+    vm.modal = null;
+
+    vm.hideModal = hideModal;
+    vm.addDish = addDish;
+    vm.deleteDish = deleteDish;
+
+    var modalScope = $rootScope.$new();
+    modalScope.ctrl = vm;
+
+    $ionicModal.fromTemplateUrl('templates/add-dish.html', {
```
Comment (dbaq) [inline app/scripts/dishes.controller.js]: The modal is never removed from the dom even when the controller is destroyed. See more at http://ionicframework.com/docs/api/service/$ionicModal/

Comment (dbaq) [inline app/scripts/dishes.controller.js]: + some lazy loading could be better

File: app/scripts/hc-messaging.service.js [hunk 4]
```diff
@@ -1,21 +1,34 @@
...
+      });
+    }
+
+    function showError(error) {
+      $ionicLoading.hide();
+      $log.error(error);
+      $ionicPopup.alert({
+        title: 'Sorry, something went wrong',
+        template: '<p align="center">' +
+        'Sorry for the inconvenience.<br><a ng-hide="showmore" href ng-click="showmore=true">show more</a>' +
```
Comment (dbaq) [inline app/scripts/hc-messaging.service.js]: I like that except that the show more link should be removed if there is no error to show.

Comment (dbaq) [inline app/scripts/hc-messaging.service.js]: href attribute can be removed.

File: app/scripts/hc-messaging.service.js [hunk 5]
```diff
@@ -1,21 +1,34 @@
...
+        template: message
+      });
+    }
+
+    function showError(error) {
+      $ionicLoading.hide();
+      $log.error(error);
+      $ionicPopup.alert({
+        title: 'Sorry, something went wrong',
+        template: '<p align="center">' +
```
Comment (dbaq) [inline app/scripts/hc-messaging.service.js]: externalize all styles to a separate scss files where you can manage all the css for hcmessaging.

File: app/scripts/hc-messaging.service.js [hunk 6]
```diff
@@ -1,21 +1,34 @@
...
+        'Sorry for the inconvenience.<br><a ng-hide="showmore" href ng-click="showmore=true">show more</a>' +
+        '<p>' +
+        '<pre ng-show="showmore" style="font-size: x-small"><code>' +
```
Comment (dbaq) [inline app/scripts/hc-messaging.service.js]: need to be externalized

**Comment by dbaq** [comment]
:+1:

File: app/scripts/hc-messaging.service.js (hunk 2 above)
Comment (valdrinkoshi) [inline app/scripts/hc-messaging.service.js]: done

File: app/scripts/hc-messaging.service.js (hunk 4 above)
Comment (valdrinkoshi) [inline app/scripts/hc-messaging.service.js]: href needed for default styling, otherwise is considered as a label

**Comment by valdrinkoshi** [comment]
@dbaq feedback incorporated, please review again

**Comment by valdrinkoshi** [comment]
will fix the merge issues now

**Comment by dbaq** [comment]
Ok looks good to me. :+1: 

I let you fix and merge.

**Pull request merged**